src/alspgrad.jl

```diff
@@ -47,7 +47,7 @@ immutable ALSGradUpdH_State{T}
     WtX::Matrix{T}    # W'X  (pre-computed)
     WtWD::Matrix{T}   # W'W * D
 
-    function ALSGradUpdH_State(X, W, H)
+    function ALSGradUpdH_State{T}(X, W, H) where T <: Real
```

Comment:
Since this package supports Julia versions prior to 0.6, the `where` syntax won't work.